hotel/models/hotel_folio.py

```diff
@@ -0,0 +1,810 @@
+     timeStamp.
+    @param server_to_client: specify timeZone offset direction (server=src
+                             and client=dest if True, or client=src and
+                             server=dest if False)
+    @param ignore_unparsable_time: if True, return False if src_tstamp_str
+                                   cannot be parsed using src_format or
+                                   formatted using dst_format.
+    @return: destination formatted timestamp, expressed in the destination
+             timezone if possible and if tz_offset is true, or src_tstamp_str
+             if timezone offset could not be determined.
+    """
+    if not src_tstamp_str:
+        return False
+    res = src_tstamp_str
+    if src_format and dst_format:
+        try:
+            # dt_value needs to be a datetime object\
+            # (so notime.struct_time or mx.DateTime.DateTime here!)
+            dt_value = datetime.strptime(src_tstamp_str, src_format)
+            if context.get("tz", False):
+                try:
+                    import pytz
+
+                    src_tz = pytz.timezone(context["tz"])
+                    dst_tz = pytz.timezone("UTC")
+                    src_dt = src_tz.localize(dt_value, is_dst=True)
+                    dt_value = src_dt.astimezone(dst_tz)
+                except Exception:
+                    pass
+            res = dt_value.strftime(dst_format)
+        except Exception:
+            # Normal ways to end up here are if strptime or strftime failed
+            if not ignore_unparsable_time:
+                return False
+            pass
+    return res
+
+
+class FolioRoomLine(models.Model):
+
+    _name = "folio.room.line"
+    _description = "Hotel Room Reservation"
+    _rec_name = "room_id"
+
+    room_id = fields.Many2one("hotel.room", "Room id")
+    check_in = fields.Datetime("Check In Date", required=True)
+    check_out = fields.Datetime("Check Out Date", required=True)
+    folio_id = fields.Many2one("hotel.folio", "Folio Number")
+    status = fields.Selection(string="state", related="folio_id.state")
+
+
+class HotelFolio(models.Model):
+
+    _name = "hotel.folio"
+    _description = "hotel folio new"
+    _rec_name = "order_id"
+    _order = "id"
+
+    def name_get(self):
+        res = []
+        fname = ""
+        for rec in self:
+            if rec.order_id:
+                fname = str(rec.name)
+                res.append((rec.id, fname))
+        return res
+
+    @api.model
+    def name_search(self, name="", args=None, operator="ilike", limit=100):
+        if args is None:
+            args = []
+        args += [("name", operator, name)]
+        folio = self.search(args, limit=100)
+        return folio.name_get()
+
+    @api.model
+    def _needaction_count(self, domain=None):
+        """
+         Show a count of draft state folio on the menu badge.
+         @param self: object pointer
+        """
+        return self.search_count([("state", "=", "draft")])
```

Comment:
I don't think, this function works after v12+

Reply:
Removed this function. This function does not work for v12+.